src/components/fileviewer.js

```diff
@@ -162,116 +137,122 @@ export default class FileViewerContainer extends Component {
+          </td>
+          <td className="file_line_text"><pre>{lineText}</pre></td>
+        </tr>
+      );
+    };
 
     return (
       <div>
         <div className="file-view">
-          <FileViewerMeta
-            revision={this.revision}
-            path={this.path}
-            status={status}
-            coverage={coverage}
-          />
-          <FileViewer
-            parsedFile={parsedFile}
-            coverage={coverage}
-            selectedLine={selectedLine}
-            onLineClick={this.setSelectedLine}
-          />
+          <FileViewerMeta parent={this} ref={c => {
+            self.status = c
+          }}/>
+          <FileViewer/>
         </div>
         <TestsSideViewer
           coverage={coverage}
-          lineNumber={selectedLine}
+          lineNumber={self.state.selectedLine}
         />
       </div>
     );
   }
 }
 
-/* This component renders each line of the file with its line number */
-const FileViewer = ({ parsedFile, coverage, selectedLine, onLineClick }) => (
-  <table className="file-view-table">
-    <tbody>
-      {
-        parsedFile.map((line, lineNumber) => (
-          <Line
-            key={lineNumber}
-            lineNumber={lineNumber + 1}
-            lineText={line}
-            coverage={coverage}
-            selectedLine={selectedLine}
-            onLineClick={onLineClick}
-          />
-        ))
-      }
-    </tbody>
-  </table>
-);
 
-const Line = ({ lineNumber, lineText, coverage, selectedLine, onLineClick }) => {
-  const handleOnClick = () => {
-    onLineClick(lineNumber);
-  };
+/* This component contains metadata of the file */
+class FileViewerMeta extends Component {
+  constructor(props) {
+    super(props);
+    this.parent = props.parent;
+    this.state = {};
+  }
 
-  const lineClass = (lineNumber === selectedLine) ? 'selected' : 'unselected';
+  render() {
+    Log.note("render status");
+    const showStatus = (label, fetched) => {
+      let msg;
+      if (fetched === undefined) {
+        msg = 'Fetching...';
+      } else if (fetched === true) {
+        msg = <span>&#x2714;</span>; // heavy checkmark
+      } else if (fetched === false) {
+        msg = <span>&#x2716;</span>; // heavy multiplication x
+      }
+      return (<li className="file-meta-li">{label}: {msg}</li>);
+    };
 
-  // default line color
-  let nTests;
-  let color = '#ffffff';
-  // hit line
-  if (coverage.coveredLines.find(element => element === lineNumber)) {
-    nTests = coverage.testsPerHitLine[lineNumber].length;
-    color = Color.getLineHitCovColor(nTests / coverage.allTests.length);
-  }
-  // miss line
-  else if (coverage.uncoveredLines.find(element => element === lineNumber)) {
-    color = '#ffe5e5';
-  }
+    return (
+      <div className="file-meta-viewer">
+        <div className="file-meta-center">
+          <div className="file-meta-title">File Coverage</div>
+          <CoveragePercentageViewer
+            coverage={this.parent.state.coverage}
```

Comment:
As per the review of the big merge; the percentage should be calculated with the rest of the coverage, and the `this.status.setState({percent: percent})`